server.js

```diff
@@ -3260,11 +3294,22 @@ function socketServer() {
                 console.log('reality editor subscription for object: ' + msgContent.object);
                 console.log('the latest socket has the ID: ' + socket.id);
 
-                realityEditorSocketArray[socket.id] = {
-                    object: msgContent.object,
-                    frame: msgContent.frame,
-                    protocol: thisProtocol
-                };
+                if (!realityEditorSocketArray[socket.id]) realityEditorSocketArray[socket.id] = [];
+
+                let isNew = true;
+                realityEditorSocketArray[socket.id].forEach((thisObj) => {
+                    if (msgContent.object === thisObj.object && msgContent.frame === thisObj.frame) {
+                        isNew = false;
+                    }
+                });
+
+                if (isNew) {
+                    realityEditorSocketArray[socket.id].push({
+                        object: msgContent.object,
+                        frame: msgContent.frame,
+                        protocol: thisProtocol
+                    });
```

Comment:
I see we're now adding the subscriptions to an array rather than only having one per socket – does this fix any bugs we were having / change any functionality? Seems like the correct approach but I'm wondering why we didn't do this before

Reply:
I'm not entirely sure. It might be working around a bug where toolsockets from the socket.io compatibility server don't appear to get unique ids but I honestly don't know. I agree that it seems like the more correct approach